Lib/Src/CDispatcher.cpp

```diff
@@ -133,54 +133,34 @@ void CDispatcher::run()
 /**
  * @brief Method to process incoming commands.
  *
- * @param command Command that arrived via coms channel
+ * @param command ICommand pointer
  * @param p_comchannel Coms channel which delivered command.
  * @return True if command has been recognised.
  */
-bool CDispatcher::newCommand(etl::string<MAX_STRING_SIZE> command,
-                             IComChannel *p_comchannel)
+bool CDispatcher::newCommand(ICommand *p_command, IComChannel *p_comchannel)
 {
     bool b_command_recognised = false;
-    // TODO: when the command API is firmed up
-    // Example. Start/stop controller.
-    // start(Controller name)/stop(Controller name)
-    etl::string<MAX_STRING_SIZE> controller_name;
-    if (command.find("stop") < command.npos)
+    const etl::string<MAX_STRING_SIZE> *command_name = p_command->getName();
+    const etl::string<MAX_STRING_SIZE> *controller_name =
+        p_command->getStringArgument();
+    if (command_name->compare("stop") == 0)
     {
-        b_command_recognised = true;
-        etl::string<MAX_STRING_SIZE>::size_type open_bracket =
-            command.find("(");
-        etl::string<MAX_STRING_SIZE>::size_type close_bracket =
-            command.find(")");
-        if ((open_bracket == command.npos) || (close_bracket == command.npos))
-        {
-            p_comchannel->send("Command: is malformatted.\n");
-        }
-        else
+        int8_t controller_number = findControllerNumber(*controller_name);
+        if (controller_number >= 0)
```

Comment:
Is it worth having another test that the returned number is not higher than actual number of controllers that were registered? It's perhaps paranoid, but if `findControllerNumber()` method was broken, then weird things will happen here. Perhaps a thought needs to be given to whether this occurence is possible.

Reply:
Probably not. Just being paranoid.